tools/c7n_azure/c7n_azure/resources/cosmos_db.py

```diff
@@ -172,6 +172,26 @@ def enumerate_resources(self, parent_resource, type_info, **params):
 
         return databases
 
+@CosmosDBDatabase.filter_registry.register('metric')
+class CosmosDBDatabaseMetricFilter(MetricFilter):
+    """CosmosDB Database Metric Filter
+    """
+    def __init__(self, data, manager=None):
+        super().__init__(data, manager)
+    
+    def get_metric_data(self, resource):
+        if self.filter is None:
+            parent_filter = "DatabaseName eq '%s'" % resource['id']
+        else:
+            parent_filter = "%s and DatabaseName eq '%s'" % (self.filter, resource['id'])
```

Comment:
I think we need add parentheses: `(%s) and (DatabaseName eq '%s')`

Reply:
Parentheses are invalid in the filter syntax: https://docs.microsoft.com/en-us/rest/api/monitor/filter-syntax
It looks like the only "complex" logic that is allowed is the "and" operator (not even "or" works), so they just assume that it is `assertion1 and assertion2 and ...`